content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1422 @@
+  // The subscribed function gets triggered whenever the backend process triggers the event 'update-ledger-index'
+  onUpdateLedgerIndex: (callback) => {
+    ipcRenderer.on('update-ledger-index', callback)
+  }
+})
+```
+
+This preloader script is used to expose functions to the browsers window object which can be used to subscribe frontend logic to events broadcast from the main logic in `index.js`.
+
+In the browser, `window.electronAPI.onUpdateLedgerIndex(callback)` can now be used tp pass a callback function via `ipcRenderer.on('eventName', callback)` that will be triggered by `appWindow.webContents.send('eventName', value)`.
+
+4. Now, in `view/template.html`, modify the line responsible for displaying the static text "Hello world!" to contain a placeholder for the ledger:index:
+
+```html
+<body>
+    <!-- Step 1 code modifications - start -->
+    <h3>Build a XRPL Wallet</h3>
+    Latest validated ledger index: <strong id="ledger-index"></strong>
+    <!-- Step 1 code modifications - end -->
+</body>
+```
+
+5. In `view/template.html` add the following line at the bottom of the file:
+
+```html
+</body>
+<!-- Step 1 code additions - start -->
+<script src="renderer.js"></script>
+<!-- Step 1 code additions - end -->
+</html>
+```
+
+6. Now create the `renderer.js` file in the `view`folder with the following code:
+
+```javascript
+const ledgerIndexEl = document.getElementById('ledger-index')
+
+// Here we define the callback function that performs the content update
+// whenever 'update-ledger-index' is called by the main process
+window.electronAPI.onUpdateLedgerIndex((_event, value) => {
+    ledgerIndexEl.innerText = value
+})
+```
+
+7. To wire up our main application to send the ledger index to the frontend, modify `index.js` by adding the following snippet:
+
+```javascript
+// Here we have to wait for the application to signal that it is ready
+// to execute our code. In this case we create a main window, query
+// the ledger for its latest index and submit the result to the main
+// window where it will be displayed
+app.whenReady().then(() => {
+  // Step 1 code additions - start
+  const appWindow = createWindow()
+  
+  getValidatedLedgerIndex().then((value) => {
+    appWindow.webContents.send('update-ledger-index', value)
+  })
+  // Step 2 code additions - end
+  
+})
+```
+
+Here we first call our helper function `getValidatedLedgerIndex()` and then broadcast an event named `update-ledger-index` and an attached payload which can be handled by the frontend. 
+
+This example shows how to do Inter Process Communication (IPC) in Electron. Technically, JavaScript has no true parallel processes or threading, because it follows a single-threaded event-driven paradigm. Nonetheless Electron provides us with two IPC modules called `ipcMain` and `ipcRenderer`. We can roughly equate those two to a backend process and a frontend process when we think in terms of client-server applications. It works as follows:
```

Comment:
Removing extra comma and spelling out which one is backend / frontend more directly
```suggestion
This example shows how to do Inter Process Communication (IPC) in Electron. Technically, JavaScript has no true parallel processes or threading because it follows a single-threaded event-driven paradigm. Nonetheless Electron provides us with two IPC modules called `ipcMain` and `ipcRenderer`. We can roughly equate `ipcMain` to a backend process and `ipcRenderer` to a frontend process when we think in terms of client-server applications. It works as follows:
```